improve check script - check zip file contents

prusnak opened the pull request:
- try to unpack zips, check file list for expected files
- add optional caching via requests_cache

I'd like to discuss the file list which is checked. Currently the `check.py` contains:

``` python
           # "LICENSE",
             "README.md",
             "__init__.py",
             "config.json",
             # "crud.py",
             # "manifest.json",
             "migrations.py",
             # "models.py",
             # "tasks.py",
             "views.py",
             # "views_api.py",
```

1. Despite what [README.md](https://github.com/lnbits/lnbits-extensions/blob/main/README.md#lighter-zip-archive) of this repo says, I think we should include and require at least `README.md` and `LICENSE` from the documentation files in the released ZIP. Text is very easy to compress and zipped shouldn't take almost any space.

2. Totally clueless about the `*.py` files. Are any of these required and necessary for the extension to function? Maybe `__init__.py`? What about the others?

3. What about `*.json` files? Is `config.json` indeed mandatory? What about `manifest.json`? These 4 extensions don't have the manifest in their repos: `smtp subdomains invoices bleskomat`. Is this an error which should be fixed or manifest is now obsolete and should be deleted from all other extensions?

dni commented:
1. ACK
2.  i really unsure about this aswell, we may not need them at  all.
3. missing manifest.json is definitly a bug!

prusnak commented:
I updated the script to check only the existence of the following files:

* `LICENSE`
* `README.md`
* `__init__.py`
* `config.json`
* `manifest.json`

I added `manifest.json` files and `LICENSE` files to repositories where they were missing, so next time the extension will be updated, the issue will be fixed.

However, I am leaving the PR in the draft state, because now it fails for all the extensions which are missing one of these 2 files.

prusnak commented:
I updated the script to check for required files (`MANDATORY_FILES` variable) and also added the list of exceptions for extensions that don't have these files in their latest release. Meanwhile I fixed all extensions on master, so this will be fixed eventually once each of these extensions publish a new release. When this is finished, we can remove all exceptions.

I think it's worth adding this check to master branch now, so at least we can perform sanity checks on newly added extensions.
